src/lib/server/locks/email-notifications.ts

```diff
@@ -62,24 +64,51 @@ async function handleEmailNotification(email: EmailNotification): Promise<void>
 	}
 }
 
-let changeStream: ChangeStream<EmailNotification>;
+let changeStream: ChangeStream<EmailNotification> | null = null;
 
-function watch() {
+async function watch(opts?: { requestChangesSince?: Timestamp }) {
 	try {
 		rateLimit('0.0.0.0', 'changeStream.email-notifications', 10, { minutes: 5 });
 	} catch {
 		console.error("Too many change streams errors for 'email-notifications', exiting");
 		process.exit(1);
 	}
-	changeStream = collections.emailNotifications.watch([{ $match: { operationType: 'insert' } }], {
-		fullDocument: 'updateLookup'
-	});
-	changeStream.on('change', (ev) => handleChanges(ev).catch(console.error));
-	changeStream.on('error', (err) => {
-		console.error('email-notifications', err);
-		changeStream.close().catch(console.error);
-		watch();
-	});
+
+	try {
+		changeStream = collections.emailNotifications
+			.watch([{ $match: { operationType: 'insert' } }], {
+				fullDocument: 'updateLookup',
+				...(opts?.requestChangesSince && {
+					startAtOperationTime: opts.requestChangesSince
+				})
+			})
+			.on('change', (ev) => handleChanges(ev).catch(console.error))
+			.once('error', async (err) => {
+				console.error('change stream error', err);
+				changeStream?.close().catch(console.error);
+				changeStream = null;
+
+				if (
+					err instanceof MongoServerError &&
+					typedInclude(['ChangeStreamHistoryLost', 'ChangeStreamFatalError'], err.codeName)
+				) {
+					// Restart from 1 hour ago if history was lost
+					watch({
+						requestChangesSince: Timestamp.fromBits(0, getUnixTime(subHours(new Date(), 1)))
+					});
+				} else {
+					watch();
```

Comment:
`return watch(...`

Else the closure will kickoff but lose the new promise; thus the promise returned to the calling function will resolve to undefined even if the underlying process has not completed.

Same in the other branch of the if expression.